packages/portfolio-contract/src/resolver/types.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * @file CCTP Resolver types - independent of portfolio-specific logic
+ *
+ * This file contains types for the CCTP transaction resolver, which is an
+ * orchestration component that can be used independently of the portfolio contract.
+ */
+
+import type { CaipChainId, AccountId } from '@agoric/orchestration';
+import type { VowKit, Vow } from '@agoric/vow';
+import { M } from '@endo/patterns';
+import type { TypedPattern } from '@agoric/internal';
+import type { Invitation } from '@agoric/zoe';
+
+/**
+ * Branded string type for CCTP transaction keys
+ * Pattern: ${AccountId}:${bigint} which expands to ${chainId}:${remoteAddress}:${amount}
+ */
+export type CCTPTransactionKey = `${AccountId}:${bigint}`;
+
+/**
+ * Details of a CCTP transaction for settlement
+ */
+export type CCTPTransactionDetails = {
+  amount: bigint;
+  /** the remoteAddress on EVM chain */
+  remoteAddress: `0x${string}`;
+  status: 'pending' | 'confirmed' | 'failed';
+};
+
+export const CCTPTransactionDetailsShape: TypedPattern<CCTPTransactionDetails> =
+  M.splitRecord(
+    {
+      amount: M.bigint(),
```

Comment:
The amount can't be negative, can it?

```suggestion
      amount: M.nat(),
```

and use `NatValue` in the type.